### PR TITLE
feat: confession

### PR DESCRIPTION
**Pull request by OysterD3**
#13

### CODE REVIEW

**Comment by yyueniao** [comment]
Thanks for your effort! Will review later.

File: .eslintrc [hunk 1]
```diff
@@ -3,72 +3,5 @@
...
-      "node": {
-        "extensions": [".js", ".ts", ".d.ts"]
-      },
-      "typescript": {
-        "project": "./tsconfig.json",
-        "alwaysTryTypes": true
-      }
-    }
-  }
+  "extends": ["@oysterlee/eslint-config-ts"]
```
Comment (yyueniao) [inline .eslintrc:6]: I am intented to use a very strict eslint rule for this project, you could disable/configure some of the rules if it is reasonable.

**Review by yyueniao — CHANGES_REQUESTED** [review]

File: .eslintrc (hunk 1 above)
Comment (yyueniao) [reply .eslintrc:6]: But I will be able to help you do the clean up and avoid the eslint warning if you are busy to do that

File: package.json [hunk 2]
```diff
@@ -34,6 +35,7 @@
     "eslint": "8.45.0",
     "eslint-config-prettier": "8.8.0",
     "eslint-plugin-import": "2.27.5",
+    "nodemon": "^3.0.1",
```
Comment (yyueniao) [inline package.json:38]: I never used this library before but looks like a great tool. Can we use exact version for dependencies as a convention?

Comment (yyueniao) [reply package.json:38]:
```suggestion
    "nodemon": "3.0.1",
```

File: src/commands/anony.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          iconURL: interaction.user.displayAvatarURL(),
+        })
+        .addFields({
+          name: ANONY_CONSTANTS.contentLabel,
+          value:
+            (interaction.options.data.find(
+              (v) => v.name === ANONY_CONSTANTS.subCommandMessageName,
+            )?.value as string) || 'No content',
+        });
+      const message = await (channel as TextChannel).send({ embeds: [embed] });
```
Comment (yyueniao) [inline src/commands/anony.ts:51]: You can avoid this type assertion by `discord.js` method `isTextBased`

Comment (yyueniao) [reply src/commands/anony.ts:51]:
```suggestion
      if (!channel.isTextBased()) {
        throw Error("Please use a text-based channel's id for `ANONYMOUS_APPROVAL_CHANNEL_ID`");
      }
      const message = await (channel as TextChannel).send({ embeds: [embed] });
```

File: src/events/reactionAdd.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        return;
+      }
+    }
+
+    if (
+      reaction.message.channel.id !== process.env.ANONYMOUS_APPROVAL_CHANNEL_ID
+    )
+      return;
+    if (reaction.message.embeds.length !== 1) return;
+    if (reaction.message.embeds[0].fields.length === 2) return; // Because reacted submission will have one extra field which is approved or rejected
```
Comment (yyueniao) [inline src/events/reactionAdd.ts:32]: IMHO, will it be better if we use `embes[0].color` to distinguish for this statement?

**Review by yyueniao — CHANGES_REQUESTED** [review]
Thank you, @OysterD3 , for your exceptional work! Please feel free to correct me if there are any inaccuracies in my reviews.

File: src/commands/anony.ts (hunk 3 above)
Comment (OysterD3) [reply src/commands/anony.ts:51]: Damn, I was wondering why this type guard won't work. Didn't know it's a method

File: src/events/reactionAdd.ts (hunk 4 above)
Comment (OysterD3) [reply src/events/reactionAdd.ts:32]: You're right

**Comment by yyueniao** [comment]
You can utilize custom script `format`, `format-fix`, `lint` and/or `lint-fix`.

**Comment by yyueniao** [comment]
I'm sorry @OysterD3 , just merged @pwkang works https://github.com/yyueniao/miters/pull/20, can you rebase and resolve the conflicts? Thanks!

File: src/commands/slash/anony.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    if (channel?.isTextBased()) {
+      const embed = new EmbedBuilder()
+        .setColor(ANONY_CONSTANTS.defaultColorCode)
+        .setTitle(
+          `(#${new Date().getTime()}) ${
+            interaction.options.data.find(
+              (data) => data.name === ANONY_CONSTANTS.subCommandTitleName,
+            )?.value ?? "Untitled"
+          }`,
+        )
```
Comment (yyueniao) [inline src/commands/slash/anony.ts:39]: May I know the reason using a timestamp as prefix? Looks unnecessary to me.
We can remove it or leave the formatted datetime at the below of the content.
```suggestion
        .setTitle(
            interaction.options.data.find(
              (data) => data.name === ANONY_CONSTANTS.subCommandTitleName,
            )?.value ?? "Untitled"
        )
```

File: tsconfig.json [hunk 6]
```diff
@@ -11,21 +24,9 @@
...
-    "moduleResolution": "node",
-
-    // Emit
-    "preserveConstEnums": true,
-    "sourceMap": true,
-
-    // Interop Constraints
-    "allowSyntheticDefaultImports": true
+    "noUnusedParameters": true
   },
```
Comment (yyueniao) [inline tsconfig.json:28]: I tried to revert to my previous configuration, everythings still work, can we change it back? If we want to update the configuration, better do it in separate PR.

**Review by yyueniao — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 6 above)
Comment (OysterD3) [reply tsconfig.json:28]: You're right, because I copied the wrong one

File: src/commands/slash/anony.ts (hunk 5 above)
Comment (yyueniao) [reply src/commands/slash/anony.ts:39]: make sense, but I will prefer to use an auto-increment id. Anyway, there is no need to add the id in this PR in my opinion(since it is big enough) let's launch it first and we can do more update in future.

Comment (OysterD3) [reply src/commands/slash/anony.ts:39]: But how would you do auto-increment id without a DB? If you say fetch by current post ID, there is a possible concurrency issue. Which risk is increasing by the amount of members

Comment (yyueniao) [reply src/commands/slash/anony.ts:39]: This could be an issue, so i prefer discuss it in separate issue and solve this in future. Currently, they can use post link or message link to mention others' post, what do you think?

Comment (OysterD3) [reply src/commands/slash/anony.ts:39]: You got a point

**Comment by yyueniao** [comment]
Well done @OysterD3 , I will invite you into collaborators, you can join us if you are interested, thanks!

**Review by yyueniao — APPROVED** [review]
